session/session.go

```diff
@@ -1499,6 +1498,9 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 	// Uncorrelated subqueries will execute once when building plan, so we reset process info before building plan.
 	cmd32 := atomic.LoadUint32(&s.GetSessionVars().CommandValue)
 	s.SetProcessInfo(stmtNode.Text(), time.Now(), byte(cmd32), 0)
+	_, digest := s.sessionVars.StmtCtx.SQLDigest()
+	s.txn.onStmtStart(digest)
```

Comment:
Seems `ExecutePreparedStmt` also needs this hook for the binary ps protocol path.

Reply:
done